ibm/resource_ibm_event_streams_schema.go

```diff
@@ -0,0 +1,355 @@
+		}
+	case "enum":
+		if err := validateName(avroType, valueString); err != nil {
+			return err
+		}
+		if _, ok := avroType["symbols"]; !ok {
+			return fmt.Errorf("%s of type %s must contain 'symbols' property", mp, valueString)
+		}
+	case "array":
+		if _, ok := avroType["items"]; !ok {
+			return fmt.Errorf("%s of type %s must contain 'items'", mp, valueString)
+		}
+	case "map":
+		if _, ok := avroType["values"]; !ok {
+			return fmt.Errorf("%s of type %s must contain 'values'", mp, valueString)
+		}
+	case "fixed":
+		if err := validateName(avroType, valueString); err != nil {
+			return err
+		}
+		if _, ok := avroType["size"]; !ok {
+			return fmt.Errorf("%s of type %s must contain 'size'", mp, valueString)
+		}
+	default:
+		if _, ok = primitiveTypes[valueString]; !ok {
+			return fmt.Errorf("unknown type: %q in schema", valueString)
+		}
+	}
+	return nil
+}
+
+func validateName(s map[string]interface{}, t string) error {
+	if _, ok := s["name"]; !ok {
+		return fmt.Errorf("missing required key-value: given schema of type %s must contain 'name'", t)
+	}
+	return nil
+}
+
+func resourceIBMEventStreamsSchemaCreate(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	schemaregistryClient, err := meta.(ClientSession).ESschemaRegistrySession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	adminURL, instanceCRN, err := getInstanceURL(d, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	schemaregistryClient.SetServiceURL(adminURL)
+	createSchemaOptions := &schemaregistryv1.CreateSchemaOptions{}
+
+	if s, ok := d.GetOk("schema"); ok {
+		var schema map[string]interface{}
+		json.Unmarshal([]byte(s.(string)), &schema)
+		createSchemaOptions.Schema = schema
+	}
+	if _, ok := d.GetOk("schema_id"); ok {
+		createSchemaOptions.SetID(d.Get("schema_id").(string))
+	}
+
+	schemaMetadata, response, err := schemaregistryClient.CreateSchemaWithContext(context, createSchemaOptions)
+	if err != nil {
+		log.Printf("[DEBUG] CreateSchemaWithContext failed %s\n%s", err, response)
+		return diag.FromErr(fmt.Errorf("CreateSchemaWithContext failed %s\n%s", err, response))
+	}
+	uniqueID := getUniqueSchemaID(instanceCRN, *schemaMetadata.ID)
+	d.SetId(uniqueID)
+
+	return resourceIBMEventStreamsSchemaRead(context, d, meta)
+}
+
+func resourceIBMEventStreamsSchemaRead(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	schemaregistryClient, err := meta.(ClientSession).ESschemaRegistrySession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	adminURL, instanceCRN, err := getInstanceURL(d, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	schemaregistryClient.SetServiceURL(adminURL)
+
+	getSchemaOptions := &schemaregistryv1.GetLatestSchemaOptions{}
+
+	schemaID := getSchemaID(d.Id())
+	getSchemaOptions.SetID(schemaID)
+
+	avroSchema, response, err := schemaregistryClient.GetLatestSchemaWithContext(context, getSchemaOptions)
+	if err != nil {
```

Comment:
Instead of checking `avroSchema ==nil `, can the response code similar to https://github.com/IBM/eventstreams-go-sdk/blob/d66f2a48a353f86ae7d04da6e9fe84be05463de5/examples/schema/main.go#L235-L237 be checked?

Reply:
`err != nil handles all kinds of response codes` it is not nil only when status is ok.. So I believe err !=nil would be enough to handle response codes..
avroSchema can be nil when there are issues in API and SDK which might result in nil pointer exception.. 
So please add `if err != nil { ==> if err != nil || avroSchema ==nil {`